pulpcore/pulpcore/app/serializers/base.py

```diff
@@ -267,24 +239,17 @@ class Meta:
     def to_representation(self, instance):
         """
         Represent a cast Detail instance as a dict of primitive datatypes
-
-        This is very similar to DRF's default to_representation implementation in
-        ModelSerializer, but makes sure to cast Detail instances and use the correct
-        serializer for rendering so that all detail fields are included. If a serializer
-        defines a 'minimal_fields' attribute, it will use these fields instead of the full set
-        of detail fields when many objects are being serialized at once (i.e. many=True,
-        when the viewset is performing a 'list' action).
         """
+
+        # This is very similar to DRF's default to_representation implementation in
+        # ModelSerializer, but makes sure to cast Detail instances and use the correct
+        # serializer for rendering so that all detail fields are included.
         ret = OrderedDict()
 
         instance = instance.cast()
         viewset = viewset_for_model(instance)()
-        serializer_class = viewset.get_serializer_class()(context=self._context)
-
-        fields = serializer_class._readable_fields
-        if self.parent and self.parent.many and hasattr(serializer_class.Meta, 'minimal_fields'):
-            fields = [field for field in fields
-                      if field.field_name in serializer_class.Meta.minimal_fields]
+        viewset.request = self._context['request']
```

Comment:
^ aforementioned hack.  not that big a deal though, and kind of unavoidable

Reply:
meh not the worst ;)